src/hooks/useUserIsVenueOwner.ts

```diff
@@ -0,0 +1,15 @@
+import { venueSelector } from "utils/selectors";
+
+import { useSelector } from "./useSelector";
+import { useUser } from "./useUser";
+
+export const useUserIsVenueOwner = (): boolean => {
+  const { user } = useUser();
+  const currentVenue = useSelector(venueSelector);
+
+  if (!currentVenue || !user) return false;
+
+  const isVenueOwner = currentVenue.owners.includes(user.uid);
+
+  return isVenueOwner;
```

Comment:
Also.. we very rarely actually have venue owners set directly on most venues. This permission is usually inherited from the 'root parent venue' (which is being called `sovereignVenue` in https://github.com/sparkletown/sparkle/pull/1047, which implements `src/hooks/useSovereignVenueId.ts`)

So the logic here is going to lock most `owners` out of being able to use this feature as they would expect.

I'm not sure exactly where/how we handle this logic currently, but it does exist in production at the moment (even without the changes in https://github.com/sparkletown/sparkle/pull/1047). It would be good to find where, and implement the same logic (ideally using some form of common helper functions rather than copy/paste replicating it)

From memory, we should check if there is a `currentVenue.parentVenueId`, if so, load that and check `owners`. If still no match, check if there is a `parentVenueId` on that, if so, load it and check `owners`, etc (repeat until you reach the 'root of the tree', which is when there is no longer a `parentVenueId`

Reply:
I expect this check would be far simpler to implement as a direct `firestore` function call within this hook (not using the redux connect patterns), properly memoised to only need to re-run when the exact dependencies have changed (in this case, that is probably `venueId` and `userId` (not `venue`/`user`, as that is 'too wide of a scope')

It could potentially even be implemented in a recursive way (though if we want to be super robust, we would probably want to keep track of which `venueId`s we've already checked so that we can't get into an infinite loop.

Reply:
I need this kind of logic to allow and prevent people from accessing the VenueAdminPage because only the owner should be able to edit the banner. And each venue has it's own banner and owners. I don't think `parentId` is relevant here. Even if the venue is a child of another venue it  should have it's own banner and different owners. Hypothetically the best way imo is to fetch all related venues merge all of the owners and check if you are there. Wdyt?

Reply:
Nope, that's not true. I don't know where in the code this is implemented currently, but the 'owners' of a 'root parent venue' are effectively 'owners' of any of those 'child venues', even though they aren't specifically in that child venue's 'owners' data.

Or said simpler: If I can admin 'root parent venue', I can admin all of it's child venues.

This happened a while back, I think it may have been a feature @thecadams worked on (at least I don't remember personally implementing it), but was basically done to reduce the 'administrative load' for setting up/running events. Instead of needing to go and add people to each venue individually, they can just be added to the one 'main' venue (usually the partymap).

Reply:
I stand corrected, will fix this.